src/nemos/basis.py

```diff
@@ -845,127 +886,136 @@ def evaluate_on_grid(self, n_samples: int) -> Tuple[NDArray, NDArray]:
-        return sample_pts * np.pi * (self.n_basis_funcs - 1)
-
     def _check_n_basis_min(self) -> None:
         """Check that the user required enough basis elements.
 
-        Check that the number of basis is at least 1.
+        Check that the number of basis is at least 2.
 
         Raises
         ------
         ValueError
-            If an insufficient number of basis element is requested for the basis type
+            If n_basis_funcs < 2.
         """
-        if self.n_basis_funcs < 1:
+        if self.n_basis_funcs < 2:
             raise ValueError(
-                f"Object class {self.__class__.__name__} requires >= 1 basis elements. "
+                f"Object class {self.__class__.__name__} requires >= 2 basis elements. "
                 f"{self.n_basis_funcs} basis elements specified instead"
             )
 
 
-class RaisedCosineBasisLog(RaisedCosineBasis):
-    """Log-spaced raised cosine basis functions used by Pillow et al. [2]_.
+class RaisedCosineBasisLog(RaisedCosineBasisLinear):
+    """Represent log-spaced raised cosine basis functions.
 
-    These are "cosine bumps" that uniformly tile the space.
+    Similar to `RaisedCosineBasisLinear` but the basis functions are log-spaced.
+    This implementation is based on the cosine bumps used by Pillow et al.[$^1$](#references)
+    to uniformly tile the internal points of the domain.
 
     Parameters
     ----------
-    n_basis_funcs
-        Number of basis functions.
+    n_basis_funcs :
+        The number of basis functions.
+    width :
+        Width of the raised cosine. By default, it's set to 2.0.
+    enforce_decay_to_zero:
+        If set to True, the algorithm first constructs a basis with `n_basis_funcs + ceil(width)` elements
+        and subsequently trims off the extra basis elements. This ensures that the final basis element
+        decays to 0.
 
     References
     ----------
-    .. [2] Pillow, J. W., Paninski, L., Uzzel, V. J., Simoncelli, E. P., & J.,
+    1. Pillow, J. W., Paninski, L., Uzzel, V. J., Simoncelli, E. P., & J.,
        C. E. (2005). Prediction and decoding of retinal ganglion cell responses
        with a probabilistic spiking model. Journal of Neuroscience, 25(47),
        11003–11013. http://dx.doi.org/10.1523/jneurosci.3305-05.2005
-
     """
 
-    def __init__(self, n_basis_funcs: int) -> None:
-        super().__init__(n_basis_funcs)
+    def __init__(
+        self,
+        n_basis_funcs: int,
+        width: float = 2.0,
+        time_scaling: float = None,
+        enforce_decay_to_zero: bool = True,
+    ) -> None:
+        super().__init__(n_basis_funcs, width=width)
+        self.enforce_decay_to_zero = enforce_decay_to_zero
+        if time_scaling is None:
+            self.time_scaling = 50.0
+        else:
+            self.time_scaling = time_scaling
+
+    @property
+    def time_scaling(self):
+        return self._time_scaling
+
+    @time_scaling.setter
+    def time_scaling(self, time_scaling):
```

Comment:
Same point as `width` about whether we want it to be settable after initialization